src/background/services/paymentManager.ts

```diff
@@ -178,21 +178,62 @@ export class PaymentManager {
   // #region One time payment
   async pay(amount: bigint, signal?: AbortSignal) {
     const payableSessions = this.payableSessions;
+    if (!payableSessions.length) {
+      throw new Error('No sessions to pay');
+    }
     this.logger.debug(`pay(${amount}) to ${payableSessions.length} sessions`);
 
-    const splitAmount = amount / BigInt(payableSessions.length);
-    const results = await Promise.allSettled(
-      payableSessions.map((session) => session.payOneTime(splitAmount)),
+    const { remainingAmount, distribution } = distributeAmount(
+      amount,
+      payableSessions,
+    );
+    if (!distribution.size) {
+      throw new Error(
+        `Cannot distribute amount (${amount}) to current sessions`,
+      );
+    }
+
+    this.logger.debug('sending outgoing payments', {
+      amount: {
+        total: amount.toString(),
+        paying: (amount - remainingAmount).toString(),
+        remaining: remainingAmount.toString(),
+      },
+      distribution: [...distribution].map(([s, amount]) => ({
+        id: s.id,
+        receiver: s.receiver.id,
+        amount: amount.toString(),
+      })),
+    });
+    const outgoingPaymentResults = await Promise.allSettled(
+      [...distribution.entries()].map(([session, amount]) =>
+        session.payOneTime(amount),
+      ),
     );
 
     this.logger.debug('polling outgoing payments for completion');
-    const result = await this.getPayStatus(results, payableSessions, signal);
+    const result = await this.getPayStatus(
+      outgoingPaymentResults,
+      Array.from(distribution.keys()),
+      signal,
+    );
 
     return {
-      amounts: {
-        ...result,
-        amount: amount.toString(),
-      },
+      amounts: { ...result, amount, remainingAmount },
+      distribution: [...distribution.entries()].map(([session, amount]) => {
+        const outgoingPayment = result.outgoingPayments.get(session.id);
+        return {
+          id: session.id,
+          walletAddress: session.receiver.id,
+          amount: amount.toString(),
+          ...(outgoingPayment
+            ? {
+                debitAmount: outgoingPayment.debitAmount.value,
+                sentAmount: outgoingPayment.sentAmount.value,
+              }
+            : {}),
+        };
+      }),
```

Comment:
It's not used right now, but can be useful with tracking (to know how well our algorithm works in practice).